massage/resynth/guitar_strummer.py

```diff
@@ -0,0 +1,275 @@
+"""Guitar Resynthesiser using chord annotation and synthesize strums
+"""
+
+import librosa
+import numpy as np
+import pretty_midi
+import scipy
+
+
+from massage.resynth import util
+from massage.core import Resynthesizer
+from massage import VOICING_FILE
+
+
+class GuitarStrummer(Resynthesizer):
+    """ Resynthesize a stereo audio file that simulate a strumming guitar.
+        Takes a guitar stem and associated JAMS object with chord annotation
+        and generate strummed chords according to JAMS chord annotation and
+        onsets detected from Audio File.
+    """
+
+    def run(self, y, fs, jam=None, instrument_label=None):
+        """ run the resynthesizer
+
+        Parameters
+        ----------
+        y : np.array
+            an audio signal.
+            shape[0] = number of channels
+            shape[1] = number of samples
+        fs : int
+            the sample rate of the audio signal y
+        jam : JAMS
+            a jams object that contain 'chord' annotation for y
+        instrument_label : str
+            Label specifying the instrument of the track. Will add more
+                -'acoustic guitar' : Acoustic Guitar
+        """
+
+        # Get Chord annotation
+        time_intervals, chord_labs = jam.search(
+            namespace='chord')[0].to_interval_values()
+
+        chord_sequence = []
+        for interval, lab in zip(time_intervals, chord_labs):
+            chord_sequence.append([interval[0], interval[1], lab])
+
+        envelope = util.get_energy_envelope(y)
+        y_mono = librosa.core.to_mono(y)
+
+        onsets, offsets, energies = util.onset_offset(y=y_mono, sr=fs)
+        velocities = util.amplitude_to_velocity(energies)
+        sound_font, program = util.pick_sf(y_mono, fs, instrument_label)
+
+        # print("soundfont: {} program: {}".format(sound_font, program))
+
+        env_times = librosa.samples_to_time(np.arange(len(envelope[0])), sr=fs)
+
+        energy_interp = []
+        # list of interp1d objects, len(.) = number of channels
+        for ch in range(envelope.shape[0]):
+            energy_interp_ch = scipy.interpolate.interp1d(
+                env_times, envelope[0], bounds_error=False,
+                fill_value='extrapolate')
+            energy_interp.append(energy_interp_ch)
+
+        # Generating midi for synthesis and also jams file
+        midi_data = self._generate_chord_midi(
+            chord_sequence, onsets, offsets, velocities, program,
+            energy_interp, VOICING_FILE)
+        # Convert midi_data -> jams file
+        jams_out = util.midi_to_jams(midi_data) # audio duration
+
+        # Resynth
+        y = midi_data.fluidsynth(sf2_path=sound_font, fs=fs)
+        # np.savez('mock_y', y)
+        if len(y) == 0:
+            print("y is none")
+            return
```

Comment:
Move this check to the beginning, and raise a `ValueError` if `len(y) == 0`. This way the function doesn't have multiple types of outputs depending on the input